tests/test_common/s3_output.py

```
@@ -13,942 +13,631 @@
+        self._max_pending_uploads = self.upload_workers * 2
+        if not self.skip_upload:
+            self._write_spool_config()
+            if self.upload_mode == "deferred":
+                self._executor = ThreadPoolExecutor(
+                    max_workers=self.upload_workers,
+                    thread_name_prefix="s3-test-log-upload",
+                )
 
+    def normalize_test_name(self, nodeid: str) -> str:
         test_name = re.sub(r"[^\w\-]", "_", nodeid)
         suffix = hashlib.md5(nodeid.encode()).hexdigest()[:8]
```

Ruff (0.15.21)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)